core/src/main/java/org/elasticsearch/common/TimeBasedUUIDGenerator.java

```diff
@@ -40,17 +41,20 @@
         assert SECURE_MUNGED_ADDRESS.length == 6;
     }
 
-    /** Puts the lower numberOfLongBytes from l into the array, starting index pos. */
-    private static void putLong(byte[] array, long l, int pos, int numberOfLongBytes) {
-        for (int i=0; i<numberOfLongBytes; ++i) {
-            array[pos+numberOfLongBytes-i-1] = (byte) (l >>> (i*8));
-        }
+    // protected for testing
+    protected long currentTimeMillis() {
+        return System.currentTimeMillis();
```

Comment:
What about initializing to System.currentTimeMillis() on startup and using that time + System.nanoTime() on generation? Regardless, if it is worth doing at all it can wait for a followup.